source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
Yes, a better comment here (and in the .cc file) would be useful. If this were new code, I'd reject it as overly vague -- what was "a problem with volatile declarations" supposed to be anyway?? :-(